modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivitySendEmailWorker.php

```diff
@@ -262,9 +275,31 @@ private function sendToFrequencyManager(array $parameters) {
 
     // We load all the target accounts.
     if ($target_accounts = $user_storage->loadMultiple($parameters['target_recipients'])) {
+      // Get the group from the activity.
+      $activity = $parameters['activity'];
+      $related_entity = $activity->getRelatedEntity();
+      if ($related_entity instanceof GroupContentInterface) {
+        /** @var \Drupal\group\Entity\GroupInterface $group */
+        $group = $related_entity->getGroup();
+      }
+      elseif ($related_entity->getEntityTypeId() === 'post') {
+        if (!$related_entity->get('field_recipient_group')->isEmpty()) {
+          /** @var \Drupal\group\Entity\GroupInterface $group */
+          $group = $related_entity->get('field_recipient_group')->first()->get('entity')->getTarget()->getValue();
+        }
+      }
+
       /** @var \Drupal\user\Entity\User $target_account */
       foreach ($target_accounts as $target_account) {
         if (($target_account instanceof User) && !$target_account->isBlocked()) {
+          // Need check if we have the group.
+          if (isset($group) && $group instanceof GroupInterface) {
+            // Only for users that didn't mute group notification.
```

Comment:
Suggestion: Skip the notification for users which have muted the group notification in which this content was posted.

Reply:
Changed.